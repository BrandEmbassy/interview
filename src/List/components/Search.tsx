import * as React from 'react';
import { changeSearchVal } from '../state';

interface Props {
  searchVal: string;
}

const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
  changeSearchVal(e.target.value);
};

const Search: React.SFC<Props> = ({ searchVal }) => {
  return (
    <div className="search">
      <input
        className="input"
        type="text"
        name="search"
        value={searchVal}
        placeholder="Search ..."
        onChange={handleSearchChange}
      />
    </div>
  );
};

export default Search;
