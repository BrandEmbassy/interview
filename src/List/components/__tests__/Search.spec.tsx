import * as React from 'react';
import Search from '../Search';
import render from '../../../test/render';

[{ searchVal: '' }, { searchVal: 'aaaa' }].forEach(props => {
  it(`should render Search - ${JSON.stringify(props, null, 2)}`, () => {
    expect(render(<Search {...props} />)).toMatchSnapshot();
  });
});
