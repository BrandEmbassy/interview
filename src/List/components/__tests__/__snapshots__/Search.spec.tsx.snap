// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search - {
  "searchVal": ""
} 1`] = `
<div
  className="search"
>
  <input
    className="input"
    name="search"
    onChange={[Function]}
    placeholder="Search ..."
    type="text"
    value=""
  />
</div>
`;

exports[`should render Search - {
  "searchVal": "aaaa"
} 1`] = `
<div
  className="search"
>
  <input
    className="input"
    name="search"
    onChange={[Function]}
    placeholder="Search ..."
    type="text"
    value="aaaa"
  />
</div>
`;
